stock_release_channel_plan_shipment_lead_time/models/stock_release_channel.py

```diff
@@ -52,3 +57,31 @@ def _compute_preparation_weekday_ids(self):
             channel.preparation_weekday_ids = self.env["time.weekday"].search(
                 [("name", "in", weekday_names)]
             )
+
+    @delivery_date_generator(step="preparation")
+    def _next_delivery_date_plan_weekdays(self, delivery_date, partner=None):
+        """Get the next valid delivery date respecting plan weekdays.
+
+        The preparation date must be a plan preparation weekday.
+        We do not consider the delivery weekday as it could be postponed with
+        leaves.
+        """
+        self.ensure_one()
+        if not self.preparation_weekday_ids:
+            while True:
+                delivery_date = yield delivery_date
```

Comment:
same as every other `delivery_date = yield delivery_date`.
You're already passing a delivery date to the generator, this looks useless and over complicated.